iwyu.cc

```diff
@@ -1728,6 +1729,15 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
         ReportDeclUse(CurrentLoc(), redecl, nullptr, UF_DefinitionUse);
       }
     }
+
+    return true;
+  }
+
+  bool VisitCleanupAttr(CleanupAttr *attr) {
+    if (CanIgnoreCurrentASTNode())
+      return true;
+
+    ReportDeclUse(CurrentLoc(), attr->getFunctionDecl());
     return true;
   }
```

Comment:
Handlers are arranged here in groups for declarations, statements, types. The better place for `VisitCleanupAttr` is outside of them, probably in a new separate group.